src/Client.php

```diff
@@ -272,7 +279,7 @@ public function request($verb, $endpoint, Session $session = null, $options = []
             throw new APIResponseException($e->getMessage(), $e->getCode(), $e);
         }
 
-        return json_decode($content);
+        return json_decode($content, $this->resultsAsArray);
```

Comment:
Hi @atymic --- I might be making some changes to this repo soon, and I can potentially incorporate this change as part of those alterations. In the mean time, you can do something like 
`$array = json_decode(json_encode($booking), true);` in order to get an assoc array from the results in your application.

Reply:
Yep, that's what I'm doing at the moment, it's just messy littering such statements around the codebase :(

Reply:
@atymic what do you think of passing it to the request method instead of a class level setting?
`    public function request($verb, $endpoint, Session $session = null, $options = [], $config = [ 'results_as' => 'object' ])
    {
        if (is_null($session) && !is_null($this->defaultSession)) {
            $session = $this->defaultSession;
        }

        if (!is_null($session)) {
            if ($session->expired()) {
                $this->refresh($session);
            }
            if (!isset($options['headers'])) {
                $options['headers'] = [];
            }
            $options['headers']['Authorization'] = "Bearer {$session->getAccessToken()}";
        }

        try {
            $response = $this->httpClient->request($verb, $endpoint, $options);

            switch($config['results_as']) {
                case 'array':
                    return json_decode($response->getBody()->getContents(), true);
                case 'response':
                    return $response;
                default:
                    return json_decode($response->getBody()->getContents());
            }
        } catch (BadResponseException $e) {
            throw new APIResponseException($e->getMessage(), $e->getCode(), $e);
        }

    }`

Reply:
IMO I would prefer to do it at a class level (it's unlikely that you'd want to use both types in the same app). You'd still have the flag in every call, and it's much easier to forget it (and cause a bug).

If you're not keen on the class level though, this would be a fine compromise.

Thanks!

Reply:
Ok fair, I'm willing to merge your suggestions if you fix that one thing I mentioned above in the constructor. I may end up adding something to the request method that allows you to override the class level setting, but it will be backwards compatible.